modules/osint/github_leaks.py

```diff
@@ -0,0 +1,132 @@
+	for term in checks:
+		url = f"{repo}/search?q={self.urlib(term).quote_plus.replace('.','%2E')}"
+		try:
+			headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36'}
+			req = self.request(url, headers=headers, timeout=20)
+		except Exception as e:
+			return
+		else:
+			try:
+				soup = BeautifulSoup(req.content, 'html.parser')
+				results = soup.find_all('div', class_='width-full')
+				if len(results) > 0:
+					for result in results:
+						OUTPUT['links'][json.loads(result.find('a')['data-hydro-click'])['payload']['result']['url']] = [re.sub(term , '\033[91m' +term+'\033[0m', line) for line in str(''.join([code.text for code in result.find_all('td' ,class_ ="blob-code blob-code-inner")])).split('\n') if term in line]
+			except:
+				return
+
+def search(self, name, q, q_formats, limit, count):
+	global PAGES,LINKS, USERS, REPO
+	engine = getattr(self, name)
+	eng = name
+	q = q_formats[f"{name}"] if f"{name}" in q_formats else q_formats['default']
+	name = engine.__init__.__name__
+	varnames = engine.__init__.__code__.co_varnames
+	if 'limit' in varnames and 'count' in varnames:
+		attr = engine(q, limit, count)
+	elif 'limit' in varnames:
+		attr = engine(q, limit)
+	else:
+		attr = engine(q)
+	if eng == 'github':
+		run = self.github(q)
+		run.run_crawl()
+		REPO += run.repositories
+	else:
+		attr.run_crawl()
+		LINKS += attr.links
+		PAGES += attr.pages
+
+
+def module_api(self):
+	global REPO
+	query = self.options['query']
+	limit = self.options['limit']
+	count = self.options['count']
+	engines = self.options['engine'].split(',')
+	q_formats = {
+		'default': f"site:github.com {query}",
+		'yippy': f'"github.com" {query}',
+		'millionshort': f'site:github.com "{query}"',
+		'qwant': f'site:github.com {query}',
+		'github': f'{query}'
+	}
+
+	self.thread(search, self.options['thread'], engines, query, q_formats, limit, count, meta['sources'])
+
+	explore = self.page_parse(PAGES).get_networks
+	bactery = ['marketplace', 'pulls', 'explore', 'issues','notifications', 'account', 'settings', 'login','about', 'pricing', 'site']
+
+	for link in self.reglib().filter(r"https://(www\.)?github\.com/[\w-]{1,39}/[\w\-\.]+", list(set(LINKS))):
+		repo = re.search(r"(https://(www\.)?github\.com/([\w-]{1,39})/[\w\-\.]+)", link)
+		if repo.group(3) not in bactery:
+			repo = repo.group(0)
+			REPO.append(repo)
+
+	REPO = list(set(REPO))
+
+	checks(self, REPO)
+
+	output = OUTPUT
+
+	self.save_gather(output,
+	 'osint/github_leaks', query, output=self.options.get('output'))
+	return output
+
+def module_run(self):
+	output = module_api(self)
+	self.alert("Links\n")
+	for url in output['links']:
+		self.output('\033[1m' + url + '\033[0m', 'G')
+		for line in output['links'][url]:
```

Comment:
Do it in the module_api if the re.sub replace is important.

Reply:
done